velocity/src/main/java/com/github/games647/fastlogin/velocity/FastLoginVelocity.java

```diff
@@ -0,0 +1,125 @@
+package com.github.games647.fastlogin.velocity;
+
+import com.github.games647.fastlogin.core.AsyncScheduler;
+import com.github.games647.fastlogin.core.CommonUtil;
+import com.github.games647.fastlogin.core.message.ChangePremiumMessage;
+import com.github.games647.fastlogin.core.message.ChannelMessage;
+import com.github.games647.fastlogin.core.message.NamespaceKey;
+import com.github.games647.fastlogin.core.message.SuccessMessage;
+import com.github.games647.fastlogin.core.shared.FastLoginCore;
+import com.github.games647.fastlogin.core.shared.PlatformPlugin;
+import com.github.games647.fastlogin.velocity.listener.ConnectListener;
+import com.github.games647.fastlogin.velocity.listener.PluginMessageListener;
+import com.google.common.collect.MapMaker;
+import com.google.common.io.ByteArrayDataOutput;
+import com.google.common.io.ByteStreams;
+import com.google.inject.Inject;
+import com.velocitypowered.api.command.CommandSource;
+import com.velocitypowered.api.event.Subscribe;
+import com.velocitypowered.api.event.player.PlayerChatEvent;
+import com.velocitypowered.api.event.proxy.ProxyInitializeEvent;
+import com.velocitypowered.api.plugin.Plugin;
+import com.velocitypowered.api.plugin.annotation.DataDirectory;
+import com.velocitypowered.api.proxy.InboundConnection;
+import com.velocitypowered.api.proxy.Player;
+import com.velocitypowered.api.proxy.ProxyServer;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentMap;
+
+import com.velocitypowered.api.proxy.messages.MinecraftChannelIdentifier;
+import com.velocitypowered.api.proxy.server.RegisteredServer;
+import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
+import org.slf4j.Logger;
+
+@Plugin(id = "fastlogin")
+public class FastLoginVelocity implements PlatformPlugin<CommandSource> {
+    private final ProxyServer server;
+    private final Path dataDirectory;
+    private final Logger logger;
+    private FastLoginCore<Player, CommandSource, FastLoginVelocity> core;
+    private final ConcurrentMap<InetSocketAddress, VelocityLoginSession> session = new MapMaker().weakKeys().makeMap();
+    private AsyncScheduler scheduler;
+
+    @Inject
+    public FastLoginVelocity(ProxyServer server, java.util.logging.Logger logger, @DataDirectory Path dataDirectory) {
```

Comment:
`java.util.Logger` is only used for backwards compatibility. If possible, we can use SL4F directly, not need to create a wrapper.

Reply:
Velocity only provides you with `java.util.logging.Logger` for logging, so I did the conversion like it's made on the bukkit and bungee version. I avoided messing around anything outside the velocity implementation to prevent making any unwanted change.

Reply:
I thought Velocity provides out of the box support for it, because the guide also indicates it: https://velocitypowered.com/wiki/developers/api-basics/

Reply:
My bad, it indeed supports it, the error it was generating was not helpful but I finally tracked down to the issue being the relocation of org.slf4j.Logger on the shade plugin which caused the inject to fail.